ZukoAzula2016/src/org/team1540/zukoazula/BangBangTalon.java

```diff
@@ -0,0 +1,63 @@
+package org.team1540.zukoazula;
+
+import ccre.channel.BooleanCell;
+import ccre.channel.BooleanInput;
+import ccre.channel.DerivedBooleanInput;
+import ccre.channel.FloatCell;
+import ccre.channel.FloatInput;
+import ccre.cluck.Cluck;
+import ccre.ctrl.ExtendedMotorFailureException;
+import ccre.drivers.ctre.talon.TalonExtendedMotor;
+import ccre.frc.FRC;
+
+public class BangBangTalon {
+    private final TalonExtendedMotor tem;
+    private final String name;
+    public final FloatInput velocity, speed;
+    public final BooleanInput isStopped, isUpToSpeed;
+
+    public BangBangTalon(TalonExtendedMotor tem, String name) {
+        this.tem = tem;
+        this.name = name;
+
+        velocity = tem.modEncoder().getEncoderVelocity();
+        Cluck.publish(name + " Velocity", velocity);
+        speed = velocity.absolute();
+        isStopped = this.speed.atMost(ZukoAzula.mainTuning.getFloat(name + " Maximum Stop Speed", 0.1f));
+        Cluck.publish(name + " Is Stopped", isStopped);
+
+        FloatInput targetSpeed = ZukoAzula.mainTuning.getFloat(name + " Target Speed", 1f);
+        FloatInput allowedVariance = ZukoAzula.mainTuning.getFloat(name + " Allowed Variance", 0.1f);
+
+        isUpToSpeed = getThreeState(velocity.atLeast(targetSpeed), velocity.atMost(targetSpeed.minus(allowedVariance.absolute())));
+        Cluck.publish(name + " At Speed", isUpToSpeed);
+    }
+
+    private static BooleanInput getThreeState(BooleanInput forceTrue, BooleanInput forceFalse) {
+        return new DerivedBooleanInput(forceTrue, forceFalse) {
+            private boolean state;
+
+            @Override
+            protected synchronized boolean apply() {
+                state |= forceTrue.get();
+                state &= !forceFalse.get();
+                return state;
+            }
+        };
+    }
+
+    private static FloatInput wrapForTests(String name, FloatInput normal) {
+        Cluck.publish(name + " Feedback", normal);
+        FloatCell testControl = new FloatCell();
+        Cluck.publish(name + " Override", testControl);
+        testControl.setWhen(0, FRC.inTestMode().onPress());
+        return FRC.inTestMode().toFloat(normal, testControl);
+    }
+
+    public void setup(BooleanInput activate) throws ExtendedMotorFailureException {
+        BooleanInput isBelowVelocity = velocity.atMost(ZukoAzula.mainTuning.getFloat(name + " Target Velocity", 1));
```

Comment:
`Target Speed` is already a value in the constructor that serves the same function. This is redundant.